### ISSUE TITLE
strange behaviour of subset function in seurat version 4.1.0

### ISSUE THREAD

**Issue by ismailelshimy** [satijalab/seurat#5815]
I noticed a strange behaviour of the subset function in  seurat version 4.1.0. I was trying to subject a large seurat object (around 27000 genes by 343000 cells). After subsetting the dataset, I got a smaller Seurat object (around 27000 genes by 90000 cells). However I noticed that the nCount_RNA values of the cells in the small data set after subsetting are completely different from the original counts in the full dataset. Also the values of the counts in the small dataset changed to become decimal numbers instead of integer counts. Did anybody notice a similar behaviour with the subset function? It is worth mentioning here that only the nCount_RNA values were changed after subsetting, the nFeature_RNA values did not change.

Example code:
full.dataset <- readRDS(file="\file_path\ file_name.rds")

subset.dataset <- subset(full.dataset, subset = variable_name == "specific value")

summary(full.dataset@meta.data$nCount_RNA)
Output:
Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    234    1318    5264   10306   13953  232128 

summary(subset.dataset@meta.data$nCount_RNA)
Output:
 Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  233.1  1350.4  2250.3  2369.9  3325.8  5060.1 

sessionInfo()
output:
R version 4.1.2 (2021-11-01)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /home/elshimyi/miniconda3/envs/CATCOVID/lib/libopenblasp-r0.3.18.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.8        purrr_0.3.4       
 [5] readr_2.1.2        tidyr_1.2.0        tibble_3.1.6       tidyverse_1.3.1   
 [9] ggplot2_3.3.5      SeuratObject_4.0.4 Seurat_4.1.0      

loaded via a namespace (and not attached):
  [1] Rtsne_0.15            colorspace_2.0-3      deldir_1.0-6         
  [4] ellipsis_0.3.2        ggridges_0.5.3        IRdisplay_1.1        
  [7] fs_1.5.2              base64enc_0.1-3       rstudioapi_0.13      
 [10] spatstat.data_2.1-4   leiden_0.3.9          listenv_0.8.0        
 [13] ggrepel_0.9.1         lubridate_1.8.0       fansi_1.0.3          
 [16] xml2_1.3.3            codetools_0.2-18      splines_4.1.2        
 [19] polyclip_1.10-0       IRkernel_1.3          jsonlite_1.8.0       
 [22] broom_0.7.12          ica_1.0-2             cluster_2.1.2        
 [25] dbplyr_2.1.1          png_0.1-7             uwot_0.1.11          
 [28] shiny_1.7.1           sctransform_0.3.3     spatstat.sparse_2.1-0
 [31] compiler_4.1.2        httr_1.4.2            backports_1.4.1      
 [34] assertthat_0.2.1      Matrix_1.4-0          fastmap_1.1.0        
 [37] lazyeval_0.2.2        cli_3.2.0             later_1.3.0          
 [40] htmltools_0.5.2       tools_4.1.2           igraph_1.2.11        
 [43] gtable_0.3.0          glue_1.6.2            RANN_2.6.1           
 [46] reshape2_1.4.4        Rcpp_1.0.8.3          scattermore_0.8      
 [49] cellranger_1.1.0      vctrs_0.3.8           nlme_3.1-155         
 [52] lmtest_0.9-40         spatstat.random_2.1-0 globals_0.14.0       
 [55] rvest_1.0.2           mime_0.12             miniUI_0.1.1.1       
 [58] lifecycle_1.0.1       irlba_2.3.5           goftest_1.2-3        
 [61] future_1.24.0         MASS_7.3-55           zoo_1.8-9            
 [64] scales_1.1.1          spatstat.core_2.4-0   hms_1.1.1            
 [67] promises_1.2.0.1      spatstat.utils_2.3-0  parallel_4.1.2       
 [70] RColorBrewer_1.1-2    reticulate_1.24       pbapply_1.5-0        
 [73] gridExtra_2.3         rpart_4.1.16          stringi_1.7.6        
 [76] repr_1.1.4            rlang_1.0.2           pkgconfig_2.0.3      
 [79] matrixStats_0.61.0    evaluate_0.15         lattice_0.20-45      
 [82] ROCR_1.0-11           tensor_1.5            patchwork_1.1.1      
 [85] htmlwidgets_1.5.4     cowplot_1.1.1         tidyselect_1.1.2     
 [88] parallelly_1.30.0     RcppAnnoy_0.0.19      plyr_1.8.7           
 [91] magrittr_2.0.2        R6_2.5.1              generics_0.1.2       
 [94] pbdZMQ_0.3-7          DBI_1.1.2             haven_2.4.3          
 [97] pillar_1.7.0          withr_2.5.0           mgcv_1.8-39          
[100] fitdistrplus_1.1-8    survival_3.3-1        abind_1.4-5          
[103] future.apply_1.8.1    modelr_0.1.8          crayon_1.5.1         
[106] uuid_1.0-4            KernSmooth_2.23-20    utf8_1.2.2           
[109] spatstat.geom_2.4-0   plotly_4.10.0         tzdb_0.3.0           
[112] readxl_1.4.0          grid_4.1.2            data.table_1.14.2    
[115] reprex_2.0.1          digest_0.6.29         xtable_1.8-4         
[118] httpuv_1.6.5          munsell_0.5.0         viridisLite_0.4.0

**Comment by yuhanH** [comment]
Hi @ismailelshimy 
The reason that `summary` results change is that overall cells are difference, so the quantile value for nCount_RNA is changed as well.  
For example, I subset 100 cells from pbmc3k. 
```
library(SeuratData)
data('pbmc3k')
pbmc3k.sub <- subset(pbmc3k, cells= Cells(pbmc3k)[1:100])
plot(pbmc3k$nCount_RNA[Cells(pbmc3k.sub)], pbmc3k.sub$nCount_RNA)
abline(a=0, b=1, col='red')

```
 `summary` results are changed, but the matched cells nCount_RNA is not changed.
```
summary(pbmc3k.sub$nCount_RNA)
 Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    688    1830    2339    2459    2884    5682 

summary(pbmc3k$nCount_RNA)
 Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    546    1756    2196    2365    2762   15818 
 ```
![image](https://user-images.githubusercontent.com/29529544/163598665-21e4119c-7d33-43a6-bdd9-1b82839da2d0.png)

But we didn't observe decimal numbers cases. Could you provide a reproducible example? Thanks.

**Comment by ismailelshimy** [comment]
@yuhanH Thank you very much for your kind reply and I apologize that I did not follow up early on your reply. I realized that the problem was actually with the original Seurat object from which I created the subset. That object had decimal count values which is related to a conversion problem from a Python to an RDS object that my collaborator who provided me with the dataset had encountered. 

Anyways, now the issue is resolved.
Thanks again @yuhanH for your kind help 😊
